<a href="https://colab.research.google.com/github/brianfarrelly/Thinkful---Data-Science/blob/master/houseprices.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [0]:
# To keep the page organized do all imports here
import pandas as pd
from scipy import stats

In [0]:
houseprices = pd.read_csv("https://tf-assets-prod.s3.amazonaws.com/tf-curric/data-science/houseprices.csv")

In [0]:
houseprices.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 81 columns):
id               1460 non-null int64
mssubclass       1460 non-null int64
mszoning         1460 non-null object
lotfrontage      1201 non-null float64
lotarea          1460 non-null int64
street           1460 non-null object
alley            91 non-null object
lotshape         1460 non-null object
landcontour      1460 non-null object
utilities        1460 non-null object
lotconfig        1460 non-null object
landslope        1460 non-null object
neighborhood     1460 non-null object
condition1       1460 non-null object
condition2       1460 non-null object
bldgtype         1460 non-null object
housestyle       1460 non-null object
overallqual      1460 non-null int64
overallcond      1460 non-null int64
yearbuilt        1460 non-null int64
yearremodadd     1460 non-null int64
roofstyle        1460 non-null object
roofmatl         1460 non-null object
exterior1st      1460 non-n

1. How many rows of data does the table contain? According to info() there are 1460 rows.


2. Are there any nulls in any columns? Yes. The alley column has only 91 non-null entries. fireplacequ has 770 non-null entries. The fence column has only 281 non-null entries. miscfeature has only 54 non-null entries. 18 columns of the 81 columns have some null entries.


3. How many distinct values are there in the mszoning column?
   5

In [0]:
houseprices['mszoning'].nunique()

5

4. How many distinct values are there in the street column?    2

In [0]:
houseprices['street'].nunique()

2

5. Describe the range of the saleprice.  34,900 - 755,000. Mean value of 180921.19. Standard Deviation of 79,442.5.  (Question: How can it be assumed that the amount is in USD?)

In [0]:
houseprices['saleprice'].describe()

count      1460.000000
mean     180921.195890
std       79442.502883
min       34900.000000
25%      129975.000000
50%      163000.000000
75%      214000.000000
max      755000.000000
Name: saleprice, dtype: float64

6. Describe the range of the lotarea.  The range in 1300-215,245. Mean value is 10,516. Std is 9981.26.

In [0]:
houseprices['lotarea'].describe()

count      1460.000000
mean      10516.828082
std        9981.264932
min        1300.000000
25%        7553.500000
50%        9478.500000
75%       11601.500000
max      215245.000000
Name: lotarea, dtype: float64

7. Is there a statistically significant difference between the prices of houses that have an open porch versus houses that do not have an open porch? How did you handle null values? Hint: use the variable openporchsf. You will need to transform this variable so that it can be used in a t-test. The rows with > 0 openporchsf were used. There is a slight positive correlation between open porch square feet and a very low p-value. So having a open porch seems to have little effect on the sale price.

In [0]:
houseprices['openporchsf'].nunique()

202

In [0]:
houseprices['openporchsf'].describe()

count    1460.000000
mean       46.660274
std        66.256028
min         0.000000
25%         0.000000
50%        25.000000
75%        68.000000
max       547.000000
Name: openporchsf, dtype: float64

In [0]:
houseprices.head(10)

id  mssubclass mszoning  ...  saletype  salecondition saleprice
0   1          60       RL  ...        WD         Normal    208500
1   2          20       RL  ...        WD         Normal    181500
2   3          60       RL  ...        WD         Normal    223500
3   4          70       RL  ...        WD        Abnorml    140000
4   5          60       RL  ...        WD         Normal    250000
5   6          50       RL  ...        WD         Normal    143000
6   7          20       RL  ...        WD         Normal    307000
7   8          60       RL  ...        WD         Normal    200000
8   9          50       RM  ...        WD        Abnorml    129900
9  10         190       RL  ...        WD         Normal    118000

[10 rows x 81 columns]

In [0]:
houseprices['openporchsf'].describe()

count    1460.000000
mean       46.660274
std        66.256028
min         0.000000
25%         0.000000
50%        25.000000
75%        68.000000
max       547.000000
Name: openporchsf, dtype: float64

In [0]:
withporch = houseprices['openporchsf'] > 0
withporch.head(10)


0     True
1    False
2     True
3     True
4     True
5     True
6     True
7     True
8    False
9     True
Name: openporchsf, dtype: bool

In [0]:
withporch.value_counts()
withporch = houseprices.iloc[withporch.values]
withporch.head(5)

id  mssubclass mszoning  ...  saletype  salecondition saleprice
0   1          60       RL  ...        WD         Normal    208500
2   3          60       RL  ...        WD         Normal    223500
3   4          70       RL  ...        WD        Abnorml    140000
4   5          60       RL  ...        WD         Normal    250000
5   6          50       RL  ...        WD         Normal    143000

[5 rows x 81 columns]

In [0]:
withporch['openporchsf'].describe()

count    804.000000
mean      84.731343
std       68.893171
min        4.000000
25%       39.000000
50%       63.000000
75%      112.000000
max      547.000000
Name: openporchsf, dtype: float64

In [0]:
withporch['openporchsf'].corr(withporch['saleprice'])

0.08645298857147718

In [0]:
stats.pearsonr(withporch['openporchsf'], withporch['saleprice'])

(0.0864529885714772, 0.014200790499984505)

8. Is there a significant correlation between the lotarea and saleprice variables? Are they positively or negatively correlated? There is a weak positive correlation between lotarea and saleprice. The data is "scattered" fairly widely. The p-value of 1.1231391549193063e-24 is insignificant and best described as zero. 
 

In [0]:
houseprices['lotarea'].describe()

count      1460.000000
mean      10516.828082
std        9981.264932
min        1300.000000
25%        7553.500000
50%        9478.500000
75%       11601.500000
max      215245.000000
Name: lotarea, dtype: float64

In [0]:
houseprices['saleprice'].describe()

count      1460.000000
mean     180921.195890
std       79442.502883
min       34900.000000
25%      129975.000000
50%      163000.000000
75%      214000.000000
max      755000.000000
Name: saleprice, dtype: float64

In [0]:
houseprices['lotarea'].corr(withporch['saleprice'])

0.2700497446740647

In [0]:
stats.pearsonr(houseprices['lotarea'], houseprices['saleprice'])

(0.2638433538714057, 1.1231391549193063e-24)